tests/unit/task_system_test.c

```diff
@@ -373,39 +386,73 @@ CTEST2(task_system, test_task_system_creation_with_bg_threads)
  * Test creation of task system using up the threads for background threads.
- * Verify that the next user-thread creation should fail.
+ * Verify ththe we can create just one more user-thread and that the next
+ * user-thread creation should fail with a proper error message.
  * ------------------------------------------------------------------------
  */
-CTEST2(task_system, test_use_all_threads_for_bg_threads)
+CTEST2(task_system, test_use_all_but_one_threads_for_bg_threads)
 {
    platform_status rc = STATUS_OK;
 
    // Destroy the task system setup by the harness, by default, w/o bg threads.
    task_system_destroy(data->hid, &data->tasks);
 
    // Consume all available threads with background threads.
-   rc = create_task_system_with_bg_threads(data, 1, (MAX_THREADS - 2));
+   rc = create_task_system_with_bg_threads(data, 1, (MAX_THREADS - 3));
    ASSERT_TRUE(SUCCESS(rc));
 
    threadid main_thread_idx = platform_get_tid();
    ASSERT_EQUAL(0, main_thread_idx, "main_thread_idx=%lu", main_thread_idx);
 
-   thread_config thread_cfg;
-   ZERO_STRUCT(thread_cfg);
-   thread_cfg.tasks                  = data->tasks;
-   thread_cfg.exp_thread_idx         = task_get_max_tid(data->tasks);
-   thread_cfg.active_threads_bitmask = task_active_tasks_mask(data->tasks);
+   thread_config_lockstep thread_cfg[2];
+   ZERO_ARRAY(thread_cfg);
+   thread_cfg[0].tasks          = data->tasks;
+   thread_cfg[0].exp_thread_idx = task_get_max_tid(data->tasks);
+   thread_cfg[0].exp_max_tid    = MAX_THREADS;
+   thread_cfg[0].line           = __LINE__;
+
+   platform_thread new_thread[2] = {0};
 
-   platform_thread new_thread = 0;
-   // This should fail to create a new thread
+   // This should successfully create a new (the last) thread
    rc = task_thread_create("test_one_thread",
-                           exec_one_thread_use_extern_apis,
-                           &thread_cfg,
+                           exec_user_thread_loop_for_stop,
+                           &thread_cfg[0],
                            trunk_get_scratch_size(),
                            data->tasks,
                            data->hid,
-                           &new_thread);
-   ASSERT_FALSE(SUCCESS(rc));
+                           &new_thread[0]);
+   ASSERT_TRUE(SUCCESS(rc));
+
+   // Wait till 1st user-thread gets to its wait-for-stop loop
+   while (!thread_cfg[0].waitfor_stop_signal) {
+      platform_sleep(USEC_TO_NSEC(100000)); // 100 msec.
+   }
+   thread_cfg[1].tasks          = data->tasks;
+   thread_cfg[1].exp_thread_idx = task_get_max_tid(data->tasks);
+
+   // We've used up all threads. This thread creation should fail.
+   rc = task_thread_create("test_one_thread",
+                           exec_user_thread_loop_for_stop,
+                           &thread_cfg[1],
+                           trunk_get_scratch_size(),
+                           data->tasks,
+                           data->hid,
+                           &new_thread[1]);
```

Comment:
The failure output from here looks like:

```
sdb-fdb-build:[48] $ build/debug/bin/unit/task_system_test test_use_all_but_one_threads_for_bg_threads
Running 1 CTests, suite name 'task_system', test case 'test_use_all_but_one_threads_for_bg_threads'.
TEST 1/1 task_system:test_use_all_but_one_threads_for_bg_threads Cannot create a new thread as the limit on concurrent threads, 64, will be exceeded.
Could not create a thread: Resource temporarily unavailable
[OK]
```

Reply:
This output shows up even when `VERBOSE` is unset. 

 In #494 I did some work to hide such "error messages from successful tests" unless the `VERBOSE` flag is set.

Please add something like this code below, either into the test case or the file's `CTEST_SETUP` function, so that the `VERBOSE` flag is respected:

https://github.com/vmware/splinterdb/blob/83de5e03722d61436ce247e406d87b8fa4913d85/tests/unit/limitations_test.c#L68-L82